### PR TITLE
Toffoli manual

### PR DESCRIPTION
**Pull request by CalMacCQ**
I have added a short section on the ``ToffoliBox`` feature to the manual. I'm happy to expand/shorten this as required.

I have also moved the section entitled "Statevectors and unitaries" to appear earlier in the manual as I mention ``Circuit.get_statevector()`` in my ``ToffoliBox`` section.

### CODE REVIEW

File: manual/manual_circuit.rst [hunk 1]
```diff
@@ -482,28 +507,35 @@ Another notable example that is common to many algorithms and high-level circuit
     circ.add_pauliexpbox(PauliExpBox([Pauli.X, Pauli.Y, Pauli.Y, Pauli.Y], 0.2), [0, 1, 2, 3])
     circ.add_pauliexpbox(PauliExpBox([Pauli.Y, Pauli.X, Pauli.Y, Pauli.Y], -0.2), [0, 1, 2, 3])
 
-Statevectors and Unitaries
---------------------------
+In addition to the box types mentioned above ``pytket`` also supports a :py:class:`ToffoliBox` structure. :py:class:`ToffoliBox` can be used to prepare an arbitrary permutation of the computational basis states using. Within the box this permutation is performed using the classical reversible gateset :math:`\{X,CnX\}`.
```
Comment (sjdilkes) [inline manual/manual_circuit.rst:510]: `can be used to prepare an arbitrary permutation of the computational basis states using.` <- It looks like you're missing the end of this sentence.

Comment (sjdilkes) [inline manual/manual_circuit.rst:510]: just say "the gateset :math:`\{X,CnX\}`.

File: manual/manual_circuit.rst [hunk 2]
```diff
@@ -482,28 +507,35 @@ Another notable example that is common to many algorithms and high-level circuit
...
+
+In order to construct a :py:class:`ToffoliBox` the user must supply the desired permutation as a dictionary specifying the action of the box on different input states. If no key:value pair is specified for a given basis state then the mapping is just the identity map and the state is unchanged by the :py:class:`ToffoliBox`.
```
Comment (sjdilkes) [inline manual/manual_circuit.rst:512]: Rewrite to be something more like this (my formatting is probably off in github!):
different input basis states, where a key:value pair given as `List[Bool]` implies that the basis state key should be mapped to the basis state value. The given dictionary should provide permutations that correspond to complete cycles of basis states, i.e. if a basis state is present as a key then it must also be present as a value in the dictionary. At construction the permutations are translated to cycles and an error is thrown if the provided permutation is invalid.

File: manual/manual_circuit.rst [hunk 3]
```diff
@@ -482,28 +507,35 @@ Another notable example that is common to many algorithms and high-level circuit
...
 
-In addition :py:meth:`Circuit.get_unitary` can be used to numerically calculate the unitary matrix that will be applied by the circuit.
+    permutation = {(0, 0): (1, 1), (1, 1): (0, 0)}       # Specify the desired permutation of the basis states
+    tb = ToffoliBox(n_qubits=2, permutation=permutation) # Construct a two qubit ToffoliBox to perform the permutation
+
+    circ = Circuit(2)              # Create a two qubit circuit
+    circ.add_toffolibox(tb, [0,1]) # Add the ToffoliBox defined above to our circuit
+    circ.get_commands()            # Display circuit commands
+
+Now lets perform a statevector calculation to ensure that the :py:class:`ToffoliBox` performs the desired permutation. Recall that when calculating the statevector ``pytket`` assumes qubits to be initialised in the :math:`|0\rangle^{\otimes n}` state.
```
Comment (sjdilkes) [inline manual/manual_circuit.rst:526]: nice!

**Review by sjdilkes — CHANGES_REQUESTED** [review]
That example is fab! 

It looks like a sentence is missing its conclusion, plus I've suggested a slightly different wording for the permutation as a dictionary.

File: manual/manual_circuit.rst (hunk 1 above)
Comment (CalMacCQ) [reply manual/manual_circuit.rst]: done

File: manual/manual_circuit.rst (hunk 2 above)
Comment (CalMacCQ) [reply manual/manual_circuit.rst]: Thanks, I've rewrote this a bit - let me know if further changes are needed.

File: manual/manual_circuit.rst (hunk 1 above)
Comment (CalMacCQ) [reply manual/manual_circuit.rst]: Thanks, done.

**Review by sjdilkes — APPROVED** [review]

**Pull request merged**